irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -115,10 +117,429 @@ namespace irods::rest
+                            return make_error_response(SYS_NOT_ALLOWED, msg.str());
+                        }
+                        else if ( status == USER_INPUT_PATH_ERR ) {
+                            return make_error_response(status, "You tried to trim a path that doesn't seem to exist in this zone.");
+                        }
+                        else if ( status == USER_INCOMPATIBLE_PARAMS ) {
+                            return make_error_response(status, "You have specified two or more parameters for your trim operation which are incompatible with each other. See https://docs.irods.org/4.3.0/icommands/user/#itrim");
+                        }
+                        else if ( status < 0 ) {
+                            return make_error_response(status, "Client experienced an error processing your trim request. Please check the logs or notify your administrator.");
+                        }
+                    }
+                }
+              } else {
+                status = rcDataObjRepl(
+                    conn(),
+                    &repl_input
+                );
+                if ( status == SYS_INVALID_FILE_PATH ) {
+                    return make_error_response(status, "You passed an invalid path for replication. Make sure that you are giving an absolute path.");
+                }
+                else if ( status == USER_INPUT_PATH_ERR ) {
+                    return make_error_response(status, "You tried to trim a path that doesn't seem to exist in this zone.");
+                }
+                else if ( status == USER_INCOMPATIBLE_PARAMS ) {
+                    return make_error_response(status, "You have specified incompatible parameters for your trim operation.");
+                }
+                else if ( status == SYS_NOT_ALLOWED ) {
+                    std::stringstream msg;
+                    msg << "You have tried to repl an object in a way that is not allowed.\n";
+                    msg << "Some possible causes are:\n";
+                    msg << "- You may have tried to replicate a non-good replica.\n";
+                    msg << "- The object you tried to replicate may have a checksum that wasn't verified.\n";
+                    msg << "- You may have tried to replicate a data object onto an already identical replica.\n";
+                    msg << "Please check for these issues and contact your administrator if the issue isn't resolved.\n";
+                    return make_error_response(SYS_NOT_ALLOWED, msg.str());
+                }
+                else if ( status < 0 ) {
+                    return make_error_response(status, "Client experienced an error processing your trim request. Please check the logs or notify your administrator.");
+                }
+              }
+
+                return std::make_tuple(
+                    Pistache::Http::Code::Ok,
+                    ""
+                );
+            }
+            catch (const irods::exception& e) {
```

Comment:
We need to catch filesystem exceptions too.

Reply:
This is marked resolved but I don't see the filesystem exception handling